Types/HttpQuery.php

```diff
@@ -35,21 +35,43 @@ class HttpQuery {
      * clause of the parsed sql tokens
      *
      * @param  array $tokens
+     * @param  array $options
      * @return string|null
      *
      * @SuppressWarnings("PHPMD.StaticAccess")
      */
-    public static function create(array $tokens) {
+    public static function create(array $tokens, array $options = []) {
         HashMap::assert($tokens, 'tokens');
 
         $operation = SqlOperation::create($tokens);
-        if ($operation !== SqlOperations::SELECT || empty($tokens['WHERE'])) return null;
+        if ($operation !== SqlOperations::SELECT) return null;
 
         $tableAlias = Table::alias($tokens);
-        $query      = array_reduce($tokens['WHERE'], function($query, $token) use ($tableAlias) {
-            return $query . str_replace('"', '', str_replace('OR', '|', str_replace('AND', '&', str_replace($tableAlias . '.', '', $token['base_expr']))));
-        });
+        
+        $query = '';
+        
+        if(isset($tokens['WHERE'])) {
```

Comment:
I'm not keen on the multiple returns here as the code is no longer DRY. I will find a happy middle ground.